### ISSUE TITLE
Unsupported operation: Undetermined nullability

### ISSUE THREAD

**Issue by llfbandit** [jodinathan/typings#2]
While experimenting, I tried to launch my Flutter app in debug mode.
As a note, when building in release mode, it seems to achieve the compilation step (runtime untested).

Flutter 3.10.5
Dart: 3.0.5

<details> 
  <summary>Here's the stack trace</summary>

```
Unhandled exception:
Unsupported operation: Undetermined nullability. Encountered while compiling file:///......./Pub/Cache/hosted/pub.dev/typings-0.0.1+1/lib/src/d/core/lib_es2015_collection_d.dart, which contains the type: InterfaceType(Object%).
lib_es2015_collection_d.dart:1
#0      ProgramCompiler._typeCompilationError (package:dev_compiler/src/kernel/compiler.dart:3355:7)
#1      ProgramCompiler._undeterminedNullabilityError (package:dev_compiler/src/kernel/compiler.dart:3352:7)
#2      ProgramCompiler._emitInterfaceType (package:dev_compiler/src/kernel/compiler.dart:3277:7)
#3      ProgramCompiler.visitInterfaceType (package:dev_compiler/src/kernel/compiler.dart:3191:7)
#4      InterfaceType.accept (package:kernel/ast.dart:12100:42)
#5      MappedListIterable.elementAt (dart:_internal/iterable.dart:415:31)
#6      MappedListIterable.elementAt (dart:_internal/iterable.dart:415:40)
#7      ListIterator.moveNext (dart:_internal/iterable.dart:344:26)
#8      InstantiatorGeneratorVisitor.splayNodes (package:dev_compiler/src/js_ast/template.dart:224:23)
#9      InstantiatorGeneratorVisitor.handleCallOrNew.<anonymous closure> (package:dev_compiler/src/js_ast/template.dart:604:22)
#10     Template.instantiate (package:dev_compiler/src/js_ast/template.dart:118:26)
#11     JsBuilder.call (package:dev_compiler/src/js_ast/builder.dart:212:21)
#12     ProgramCompiler._emitGenericClassType (package:dev_compiler/src/kernel/compiler.dart:3393:15)
#13     ProgramCompiler._emitInterfaceType (package:dev_compiler/src/kernel/compiler.dart:3256:17)
#14     ProgramCompiler.visitInterfaceType (package:dev_compiler/src/kernel/compiler.dart:3191:7)

#15     InterfaceType.accept (package:kernel/ast.dart:12100:42)
#16     ProgramCompiler._emitType (package:dev_compiler/src/kernel/compiler.dart:3076:60)
#17     ProgramCompiler._emitArgumentList (package:dev_compiler/src/kernel/compiler.dart:6223:41)
#18     ProgramCompiler._emitMethodCall (package:dev_compiler/src/kernel/compiler.dart:5326:16)
#19     ProgramCompiler.visitInstanceInvocation (package:dev_compiler/src/kernel/compiler.dart:5240:12)
#20     InstanceInvocation.accept (package:kernel/ast.dart:6128:44)
#21     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4232:20)
#22     ProgramCompiler._emitMethodCall (package:dev_compiler/src/kernel/compiler.dart:5325:22)
#23     ProgramCompiler.visitInstanceInvocation (package:dev_compiler/src/kernel/compiler.dart:5240:12)
#24     InstanceInvocation.accept (package:kernel/ast.dart:6128:44)
#25     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4232:20)
#26     ProgramCompiler.visitConditionalExpression (package:dev_compiler/src/kernel/compiler.dart:6598:21)
#27     ConditionalExpression.accept (package:kernel/ast.dart:7420:44)
#28     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4232:20)
#29     ProgramCompiler.visitLet (package:dev_compiler/src/kernel/compiler.dart:6996:16)
#30     Let.accept (package:kernel/ast.dart:9277:44)
#31     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4232:20)
#32     ProgramCompiler.visitLet (package:dev_compiler/src/kernel/compiler.dart:6995:16)
#33     Let.accept (package:kernel/ast.dart:9277:44)
#34     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4232:20)
#35     MappedListIterable.elementAt (dart:_internal/iterable.dart:415:31)
#36     ListIterator.moveNext (dart:_internal/iterable.dart:344:26)

#37     new _GrowableList._ofEfficientLengthIterable (dart:core-patch/growable_array.dart:189:27)
#38     new _GrowableList.of (dart:core-patch/growable_array.dart:150:28)
#39     new List.of (dart:core-patch/array_patch.dart:47:28)
#40     ListIterable.toList (dart:_internal/iterable.dart:214:7)
#41     ProgramCompiler._visitExpressionList (package:dev_compiler/src/kernel/compiler.dart:4195:40)
#42     ProgramCompiler.visitListLiteral (package:dev_compiler/src/kernel/compiler.dart:6833:20)
#43     ListLiteral.accept (package:kernel/ast.dart:8700:44)
#44     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4232:20)
#45     ProgramCompiler._emitArgumentList (package:dev_compiler/src/kernel/compiler.dart:6232:11)

#46     ProgramCompiler.visitStaticInvocation (package:dev_compiler/src/kernel/compiler.dart:6146:16)
#47     StaticInvocation.accept (package:kernel/ast.dart:7029:44)
#48     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4232:20)
#49     ProgramCompiler.visitReturnStatement (package:dev_compiler/src/kernel/compiler.dart:4794:45)
#50     ReturnStatement.accept (package:kernel/ast.dart:10786:43)
#51     ProgramCompiler._visitStatement (package:dev_compiler/src/kernel/compiler.dart:4141:20)

#52     ProgramCompiler._emitFunctionScopedBody (package:dev_compiler/src/kernel/compiler.dart:4174:18)
#53     ProgramCompiler._emitSyncFunctionBody.<anonymous closure> (package:dev_compiler/src/kernel/compiler.dart:3909:17)
#54     ProgramCompiler._withLetScope (package:dev_compiler/src/kernel/compiler.dart:2593:25)

#55     ProgramCompiler._withCurrentFunction (package:dev_compiler/src/kernel/compiler.dart:3943:18)
#56     ProgramCompiler._emitSyncFunctionBody (package:dev_compiler/src/kernel/compiler.dart:3905:17)

#57     ProgramCompiler._emitFactoryConstructor (package:dev_compiler/src/kernel/compiler.dart:2310:18)
#58     ProgramCompiler._emitJSInteropClassNonExternalMembers (package:dev_compiler/src/kernel/compiler.dart:919:23)

#59     ProgramCompiler._emitClass (package:dev_compiler/src/kernel/compiler.dart:734:31)
#60     List.forEach (dart:core-patch/growable_array.dart:416:8)
#61     ProgramCompiler._emitLibrary (package:dev_compiler/src/kernel/compiler.dart:670:23)
#62     List.forEach (dart:core-patch/growable_array.dart:416:8)

#63     ProgramCompiler.emitModule (package:dev_compiler/src/kernel/compiler.dart:479:15)
#64     IncrementalJavaScriptBundler.compile (package:frontend_server/src/javascript_bundle.dart:217:33)
#65     FrontendCompiler.writeJavaScriptBundle (package:frontend_server/frontend_server.dart:778:46)
<asynchronous suspension>
#66     FrontendCompiler.compile (package:frontend_server/frontend_server.dart:641:9)
<asynchronous suspension>

#67     listenAndCompile.<anonymous closure> (package:frontend_server/frontend_server.dart:1308:11)
<asynchronous suspension>

the Dart compiler exited unexpectedly.
Failed to compile application.

```</details>

**Comment by deepak786** [comment]
Any update on this issue?

**Comment by jodinathan** [comment]
it is a bug in Dart.

The Dart team found the culprit and is working on it: https://github.com/dart-lang/sdk/issues/52726

**Comment by deepak786** [comment]
I just migrated from js_bindings to typings. Because js_bindings is not working with Flutter 3.13
But now end up having this issue.

**Comment by jodinathan** [comment]
@deepak786 I have commited a fix in master, can you try and let me know if it works for you?

**Comment by deepak786** [comment]
```
typings:
    git:
      url: https://github.com/jodinathan/typings
      path: typings
      ref: 5c5c706e29be4ed535e4b787198e746202bc2f08
```

Still getting the same error:
```
Unhandled exception:
Unsupported operation: Undetermined nullability. Encountered while compiling file:///Users/appleapple/.pub-cache/git/typings-5c5c706e29be4ed535e4b787198e746202bc2f08/typings/lib/src/d/core/lib.es2015.collection.d.dart, which contains the type: InterfaceType(Object%).
#0      ProgramCompiler._typeCompilationError (package:dev_compiler/src/kernel/compiler.dart:3509:7)
#1      ProgramCompiler._undeterminedNullabilityError (package:dev_compiler/src/kernel/compiler.dart:3506:7)
#2      ProgramCompiler._emitInterfaceType (package:dev_compiler/src/kernel/compiler.dart:3431:7)
#3      ProgramCompiler.visitInterfaceType (package:dev_compiler/src/kernel/compiler.dart:3346:7)
#4      InterfaceType.accept (package:kernel/ast.dart:11385:42)
#5      MappedListIterable.elementAt (dart:_internal/iterable.dart:415:31)
#6      MappedListIterable.elementAt (dart:_internal/iterable.dart:415:40)
#7      ListIterator.moveNext (dart:_internal/iterable.dart:344:26)
#8      InstantiatorGeneratorVisitor.splayNodes (package:dev_compiler/src/js_ast/template.dart:224:23)
#9      InstantiatorGeneratorVisitor.handleCallOrNew.<anonymous closure> (package:dev_compiler/src/js_ast/template.dart:604:22)
#10     Template.instantiate (package:dev_compiler/src/js_ast/template.dart:118:26)
#11     JsBuilder.call (package:dev_compiler/src/js_ast/builder.dart:212:21)
#12     ProgramCompiler._emitGenericClassType (package:dev_compiler/src/kernel/compiler.dart:3547:15)
#13     ProgramCompiler._emitInterfaceType (package:dev_compiler/src/kernel/compiler.dart:3410:17)
#14     ProgramCompiler.visitInterfaceType (package:dev_compiler/src/kernel/compiler.dart:3346:7)
#15     InterfaceType.accept (package:kernel/ast.dart:11385:42)
#16     ProgramCompiler._emitType (package:dev_compiler/src/kernel/compiler.dart:3231:60)
#17     ProgramCompiler._emitArgumentList (package:dev_compiler/src/kernel/compiler.dart:6406:41)
#18     ProgramCompiler._emitMethodCall (package:dev_compiler/src/kernel/compiler.dart:5476:16)
#19     ProgramCompiler.visitInstanceInvocation (package:dev_compiler/src/kernel/compiler.dart:5390:12)
#20     InstanceInvocation.accept (package:kernel/ast.dart:5770:44)
#21     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4386:20)
#22     ProgramCompiler._emitMethodCall (package:dev_compiler/src/kernel/compiler.dart:5475:22)
#23     ProgramCompiler.visitInstanceInvocation (package:dev_compiler/src/kernel/compiler.dart:5390:12)
#24     InstanceInvocation.accept (package:kernel/ast.dart:5770:44)
#25     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4386:20)
#26     ProgramCompiler.visitConditionalExpression (package:dev_compiler/src/kernel/compiler.dart:6784:21)
#27     ConditionalExpression.accept (package:kernel/ast.dart:6935:44)
#28     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4386:20)
#29     ProgramCompiler.visitLet (package:dev_compiler/src/kernel/compiler.dart:7198:16)
#30     Let.accept (package:kernel/ast.dart:8715:44)
#31     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4386:20)
#32     ProgramCompiler.visitLet (package:dev_compiler/src/kernel/compiler.dart:7197:16)
#33     Let.accept (package:kernel/ast.dart:8715:44)
#34     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4386:20)
#35     MappedListIterable.elementAt (dart:_internal/iterable.dart:415:31)
#36     ListIterator.moveNext (dart:_internal/iterable.dart:344:26)
#37     new _GrowableList._ofEfficientLengthIterable (dart:core-patch/growable_array.dart:189:27)
#38     new _GrowableList.of (dart:core-patch/growable_array.dart:150:28)
#39     new List.of (dart:core-patch/array_patch.dart:47:28)
#40     ListIterable.toList (dart:_internal/iterable.dart:214:7)
#41     ProgramCompiler._visitExpressionList (package:dev_compiler/src/kernel/compiler.dart:4349:40)
#42     ProgramCompiler.visitListLiteral (package:dev_compiler/src/kernel/compiler.dart:7025:20)
#43     ListLiteral.accept (package:kernel/ast.dart:8161:44)
#44     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4386:20)
#45     ProgramCompiler._emitArgumentList (package:dev_compiler/src/kernel/compiler.dart:6415:11)
#46     ProgramCompiler.visitStaticInvocation (package:dev_compiler/src/kernel/compiler.dart:6316:16)
#47     StaticInvocation.accept (package:kernel/ast.dart:6578:44)
#48     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:4386:20)
#49     ProgramCompiler.visitReturnStatement (package:dev_compiler/src/kernel/compiler.dart:4941:45)
#50     ReturnStatement.accept (package:kernel/ast.dart:10127:43)
#51     ProgramCompiler._visitStatement (package:dev_compiler/src/kernel/compiler.dart:4295:20)
#52     ProgramCompiler._emitFunctionScopedBody (package:dev_compiler/src/kernel/compiler.dart:4328:18)
#53     ProgramCompiler._emitSyncFunctionBody.<anonymous closure> (package:dev_compiler/src/kernel/compiler.dart:4063:17)
#54     ProgramCompiler._withLetScope (package:dev_compiler/src/kernel/compiler.dart:2693:25)
#55     ProgramCompiler._withCurrentFunction (package:dev_compiler/src/kernel/compiler.dart:4097:18)
#56     ProgramCompiler._emitSyncFunctionBody (package:dev_compiler/src/kernel/compiler.dart:4059:17)
#57     ProgramCompiler._emitFactoryConstructor (package:dev_compiler/src/kernel/compiler.dart:2410:18)
#58     ProgramCompiler._emitJSInteropClassNonExternalMembers (package:dev_compiler/src/kernel/compiler.dart:991:23)
#59     ProgramCompiler._emitClass (package:dev_compiler/src/kernel/compiler.dart:804:31)
#60     List.forEach (dart:core-patch/growable_array.dart:416:8)
#61     ProgramCompiler._emitLibrary (package:dev_compiler/src/kernel/compiler.dart:740:23)
#62     List.forEach (dart:core-patch/growable_array.dart:416:8)
#63     ProgramCompiler.emitModule (package:dev_compiler/src/kernel/compiler.dart:491:15)
#64     IncrementalJavaScriptBundler.compile (package:frontend_server/src/javascript_bundle.dart:220:33)
#65     FrontendCompiler.writeJavaScriptBundle (package:frontend_server/frontend_server.dart:786:46)
<asynchronous suspension>
#66     FrontendCompiler.compile (package:frontend_server/frontend_server.dart:650:9)
<asynchronous suspension>
#67     listenAndCompile.<anonymous closure> (package:frontend_server/frontend_server.dart:1316:11)
<asynchronous suspension>
Finished with error: the Dart compiler exited unexpectedly.
Failed to compile application.
```

**Comment by jodinathan** [comment]
is the ref correct to use the last commit? I think you can remove the ref property just to make sure you are fetching from latest version.  

If it doesn't work we will have to wait for the Dart team to fix it because I can't find the culprit.  

I've asked in the issue if it will be fixed in 3.2.

**Comment by deepak786** [comment]
I tried same issue.
Also tried by overriding the ts2dart

```
dependency_overrides:
  ts2dart:
    git:
      url: https://github.com/jodinathan/typings
      path: ts2dart
```

Still same issue.

**Comment by deepak786** [comment]
I have updated the Flutter to 3.16.2 which has Dart 3.2.2
Now it works fine. So it seems that the issue is fixed in Dart 3.2.2

**Comment by jodinathan** [comment]
fixed with Dart 3.2.2
